kubernetes/resource_kubernetes_service_account.go

```diff
@@ -104,11 +105,17 @@ func resourceKubernetesServiceAccountCreate(d *schema.ResourceData, meta interfa
 	})
 
 	diff := diffObjectReferences(svcAcc.Secrets, resp.Secrets)
+	defaultSecret := diff[0]
 	if len(diff) > 1 {
-		return fmt.Errorf("Expected 1 generated default secret, %d found: %s", len(diff), diff)
+		for _, secret := range diff {
+			if strings.Contains(secret.Name, "-token-") {
```

Comment:
This approach is problematic because there is no API contract to guarantee that token secrets must have the `-token-` string will be part of the name for token secrets.

There is however a `Type` attribute on `Secret` objects (see [here](https://github.com/kubernetes/api/blob/173ce66c1e39d1d0f56e0b3347ff2988068aecd0/core/v1/types.go#L4942)). You can filter the list of Secrets on this `Type` attribute for all objects with a value of `kubernetes.io/service-account-token` and that should give you a list of just the token type secrets. You'll most likely just get one during create, but I guess you can just pick the first one if you get more (very non-deterministic unfortunately).

Reply:
Filtering on the secret type is indeed much cleaner!

Reply:
I added the filtering on secret type. I think we can keep the previous logic and return an error if we have more than one service account token. The filtering applies even if there is only one secret, I'm not sure if it's what you want... And we possibly can move this part in the "resource.Retry" block if we want to wait for a service account token to appear ?

Reply:
@alexsomesan ?

Reply:
@pablo-ruth I think adding the `resource.Retry` around it is a good idea. 

However, I'm a bit concerned about doing `.Get(..)` calls for individual objects in a loop (the `for...range` around it) AND wrapping that in retries. It has the theoretical potential to cause a lot of requests in worst case conditions (and stress the API server unnecessarily).
The alternative would be to use a `.Secrets(namespace).List(..)` type of call OUTSIDE the loop and then filter it's results for the relevant secret name. We can that wrap that in a retry and not have it quadratically grow out of control.

What do you think? If you'd rather not add the retry, than I think we can merge like this.

Reply:
Thanks!